### DISCUSSION TITLE
compiling with PETSc 3.18

### DISCUSSION THREAD

**Discussion by garciapintado** [idaholab/moose#22565] [Q&A Getting Started] [answered]
Hello,
I'm trying to compile moose in a HPC cluster [single user] following:
https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html

 And I'm finding an error message with the indication:
>
> **************************ERROR*************************************
>
> Error during compile, check arch-moose/lib/petsc/conf/make.log
>
> Send it and arch-moose/lib/petsc/conf/configure.log to
> petsc-maint@mcs.anl.gov
>
> ********************************************************************

With this, the reponse I've got from a PETSc developer is:

--
Dear Javier,
This has been fixed in later releases of PETSc.
The latest to date is version 3.18.1, could you try that instead?

Thanks,
Pierre
--

But on the other hand, I understand that MOOSE is sort of fixed to a specific PETSc version (3.16). Is this so? If not, please could you guide on how compile MOOSE with a more recent PETSc version?
Best
Javier
...

**Comment by milljm** [comment]
I would take a look at the script we use to configure PETSc: `moose/scripts/configure_petsc.sh`. This is the script that is called by another script, the script we tell everyone to run: `update_and_rebuild_petsc.sh`. Who's purpose is to obtain the PETSc source at a version we support. But I like the journey you're on, so lets continue!

Of course I can't promise all of the same arguments will work with 3.18, but they probably will. Also, some tests my fail at this version. And in-turn may incorrectly solve problems. Which is why it's best to stick with an older stable (supported) version.

### The gist of what you need to do:

- obtain PETSc 3.18 source
- Extract it (un-tar it)
- Enter the directory and run
  ```bash
  python ./configure --download-hypre=1 \
      --with-shared-libraries=1 \
      --download-hdf5=1 \
      --download-hdf5-fortran-bindings=0 \
      --with-make-np=6 \
      --with-debugging=no \
      --download-fblaslapack=1 \
      --download-metis=1 \
      --download-ptscotch=1 \
      --download-parmetis=1 \
      --download-superlu_dist=1 \
      --download-mumps=1 \
      --download-strumpack=1 \
      --download-scalapack=1 \
      --download-slepc=1 \
      --with-mpi=1 \
      --with-openmp=1 \
      --with-cxx-dialect=C++11 \
      --with-fortran-bindings=0 \
      --with-sowing=0 \
      --with-64-bit-indices \
      --prefix=/SOME/PATH/WHERE/YOU/WANT/TO/INSTALL
  ```
Obviously change that last bit, to where you plan on installing this version of PETSc. Whatever this path is, you will need to set `export PETSC_DIR=<the path you used>`, before attempting to build libMesh.

I hope this works for you! You're exploring new territory 😃  Please let us know how it goes either way!

**Comment by loganharbour** [comment, marked as answer]
I believe https://github.com/idaholab/moose/pull/22562 should satisfy this issue. While it says PETSc 3.8, I believe this update is for PETSc 3.18 compatibility.

**Reply by garciapintado** [reply]
Ah! I see. Then it seems sensible for me to wait until #22562 is solved...
